Handle Quaternionf and Vector3f in entity metadata

mat-1 opened the pull request:
The new snapshot added a new "display" entity which contains metadata fields that are of type org.joml.Vector3f and org.joml.Quaternionf. Right now burger errors and fails to generate entity metadata since it doesn't know how to handle those, this fixes that.

mat-1 committed:
Handle Quaternionf and Vector3f in entity metadata

mat-1 committed:
change some ifs to elifs

Pokechu22 commented:
This ends up looking a bit weird in the output:

![image](https://user-images.githubusercontent.com/8334194/217660126-15b3ed13-db4c-49d1-a0de-ce86e30f8706.png)

I'm not sure what would make it better, though; making it a string might help or might not. (Also, as far as I can tell, the existing text component one isn't used anymore; I'm not sure if it was ever used or if it was always broken, though).

Adding something like this:

```python
elif inner_type.startswith("org/joml/"):
    name = inner_type[len("org/joml/"):]
```

to `_serializer_name` would give them correct names fairly easily; I figured that part out but hadn't figured out why it wasn't working in general.

mat-1 committed:
quaternionf/vector3f serializer names

mat-1 commented:
Alright, I changed it to have a serializer name there. That was actually also the first thing I did to try to fix it, but when it didn't I assumed it was irrelevant and removed it, lol.
